lib/puppet/provider/ec2_instance/v2.rb

```diff
@@ -70,6 +71,7 @@ def self.instance_to_hash(region, instance)
       tags: tags,
       region: region,
       hypervisor: instance.hypervisor,
+      iam_instance_profile_arn: instance.iam_instance_profile ? instance.iam_instance_profile.arn : nil,
```

Comment:
The instance response returned from the amazon api does not provide the IAM role friendly name.

I have looked at the Aws::IAM::Client to see if I can get the friendly name but have been unable to find a method.  The closest I could work out (which may be over kill) would be to get a list of all the roles the user can query against and than look up the arn returned to find the friendly name.

Something like:
```ruby
m_iam = Aws::IAM::Client.new(region:'ap-southeast-2')
iam_instance_profile_arn = m_iam.list_instance_profiles.instance_profiles.select {|item| item.arn == 'arn:aws:iam::#########:instance-profile/???????????'}.map(&:instance_profile_name)
```
Which is not ideal.

Thoughts?